arches/app/views/api.py

```diff
@@ -2149,69 +2149,55 @@ def get(self, request, identifier=None):
 
         return JSONResponse(response_data)
 
-    def attribute_nodes_are_valid(self, geom_node=models.Node, attribute_nodeids=[]):
-        attribute_node_graphs = [
-            n.graph_id for n in models.Node.objects.filter(nodeid__in=attribute_nodeids)
-        ]
-        # if any of the graphs in attribute_node_graphs != geom_node.graph return False
-        are_valid = not any(
-            graph_id != geom_node.graph_id for graph_id in attribute_node_graphs
-        )
-
-        return are_valid
-
-    def identifier_is_uuid(self, identifier):
-        uuid_pattern = re.compile(settings.UUID_REGEX, re.IGNORECASE)
-        return uuid_pattern.match(identifier)
-
-    def create_json_data_object_from_spatialview(self, spatialview):
+    def transform_json_data_for_spatialview(self, json_data):
         """
-        Returns a JSON object representing the spatialview
+        Transforms the JSON data object to be used in the spatialview model
         """
-        return {
-            "spatialviewid": str(spatialview.spatialviewid),
-            "schema": spatialview.schema,
-            "slug": spatialview.slug,
-            "description": spatialview.description,
-            "geometrynodeid": str(spatialview.geometrynode.pk),
-            "ismixedgeometrytypes": spatialview.ismixedgeometrytypes,
-            "language": spatialview.language.code,
-            "attributenodes": spatialview.attributenodes,
-            "isactive": spatialview.isactive,
-        }
+        json_data["geometrynode"] = models.Node.objects.get(
+            nodeid=json_data["geometrynodeid"]
+        )
+        del json_data["geometrynodeid"]
```

Comment:
Setting the field_id should achieve the same thing in a lazy way (no db query):
```suggestion
        json_data["geometrynode_id"] = json_data.pop("geometrynodeid")
```

Reply:
Neat trick! Works a charm